onedal/basic_statistics/basic_statistics.py

```diff
@@ -63,59 +62,67 @@ def _get_onedal_params(self, is_csr, dtype=np.float32):
             "result_option": options,
         }
 
-    def _compute_raw(
-        self, data_table, weights_table, module, policy, dtype=np.float32, is_csr=False
-    ):
-        params = self._get_onedal_params(is_csr, dtype)
-
-        result = module.train(policy, params, data_table, weights_table)
 
-        options = self._get_result_options(self.options)
-        options = options.split("|")
+class BasicStatistics(BaseBasicStatistics):
+    """
+    Basic Statistics oneDAL implementation.
+    """
 
-        return {opt: getattr(result, opt) for opt in options}
+    def __init__(self, result_options="all", algorithm="by_default"):
+        super().__init__(result_options, algorithm)
 
-    def _compute(self, data, weights, module, queue):
-        policy = self._get_policy(queue, data, weights)
+    def fit(self, data, sample_weight=None, queue=None):
+        policy = self._get_policy(queue, data, sample_weight)
 
         is_csr = _is_csr(data)
-        if not (data is None) and not is_csr:
-            data = np.asarray(data)
-
-        if not (weights is None):
-            weights = np.asarray(weights)
 
-        data, weights = _convert_to_supported(policy, data, weights)
+        if not (data is None) and not is_csr:
```

Comment:
```suggestion
        if data is not None and not is_csr:
```
Generally this would be preferrable for python, there are a few places where it can be changed